onedal/covariance/incremental_covariance.py

```diff
@@ -114,19 +115,24 @@ def finalize_fit(self, queue=None):
         Parameters
         ----------
         queue : dpctl.SyclQueue
-            Not used here, added for API conformance
+            If not None, use this queue for computations.
 
         Returns
         -------
         self : object
             Returns the instance itself.
         """
         params = self._get_onedal_params(self._dtype)
+        if queue is not None:
+            policy = self._get_policy(queue)
+        else:
+            policy = self._get_policy(self._queue)
+
```

Comment:
Logic in finalize_fit and partial_fit does not match - can we align this? In terms of getting policy from queue/self._queue

Reply:
it shouldn't match. in partial_fit queue is supposed to be provided explicitly from sklearnex side via data. finalize_fit is designed to handle the case queue is not provided. it takes it from self._queue in this case assuming the same device as for partial_fit calls must be used .